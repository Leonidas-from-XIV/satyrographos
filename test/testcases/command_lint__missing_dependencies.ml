open Core
open Satyrographos_testlib

let satysfi_package_opam =
  "satysfi-package.opam", TestLib.opam_file_for_test
    ~name:"satysfi-package"
    ~version:"0.1"
    ()

let satysfi_package_doc_opam =
  "satysfi-package-doc.opam", TestLib.opam_file_for_test
    ~name:"satysfi-package-doc"
    ~version:"0.1"
    ()

let satyristes =
  "Satyristes", sprintf
    {|(version "0.0.2")

(library
  (name "package")
  (version "0.1")
  (sources ())
  (opam "satysfi-package.opam")
  (dependencies ((unspecified-package ()))))

(libraryDoc
  (name "package-doc")
  (version "0.1")
  (build ())
  (sources ())
  (opam "satysfi-package-doc.opam")
  (dependencies ((package ()))))
|}

let files =
  [ satysfi_package_opam;
    satysfi_package_doc_opam;
    satyristes;
  ]

let opam_libs = Satyrographos.Library.[
    {empty with name = Some "package"};
    {empty with name = Some "unspecified-package"};
  ]

let () =
  TestCommand.test_lint_command ~opam_libs files
