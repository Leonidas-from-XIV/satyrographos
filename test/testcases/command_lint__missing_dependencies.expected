@@test_library@@/Satyristes:3:1: (module package):
@@test_library@@/satysfi-package.opam:
Warning:
  The OPAM file lacks dependencies on specified SATySFi libraries: (unspecified-package).

@@test_library@@/Satyristes:10:1: (module package-doc):
@@test_library@@/satysfi-package-doc.opam:
Warning:
  The OPAM file lacks dependencies on specified SATySFi libraries: (package).

2 problem(s) found.
